### PR TITLE
Add placeholder for JxBrowser dependency

### PR DESCRIPTION
**Pull request by djlan**

### CODE REVIEW

File: Utils/azuretools-core/pom.xml [hunk 1]
```diff
@@ -184,5 +184,10 @@
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>18.0</version>
```
Comment (voidfunction) [inline Utils/azuretools-core/pom.xml]: why we have version information here?

Comment (djlan) [reply Utils/azuretools-core/pom.xml]: Removed

---
In reply to: [154040202](https://github.com/Microsoft/azure-tools-for-java/pull/1279#discussion_r154040202) [](ancestors = 154040202)

File: Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserUtil.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ *  TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ */
+
+package com.microsoft.azuretools.ui.embeddedbrowser;
+
+import com.google.common.collect.ImmutableMap;
+
+public class JxBrowserUtil {
+    private static final ImmutableMap<String, String> versionMap = ImmutableMap.of(
```
Comment (wezhang) [inline Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserUtil.java]:
```
diff /home/jenkins/dependencyLog//develop/intellij/528-intellij.log /home/jenkins/dependencyLog//PR/intellij/1279-intellij.log
+ cat /home/jenkins/dependencyLog//patch/PR-1279.patch
+ wc -l
+ [ 5 -eq 0 ]
+ cat /home/jenkins/dependencyLog//patch/PR-1279.patch
135c135,136
< |    \--- com.google.code.gson:gson:2.7
---
> |    +--- com.google.code.gson:gson:2.7
> |    \--- com.google.guava:guava:20.0
+ echo ALERT: New dependencies introduced!
```

You could try Java `java.util.AbstractMap.SimpleImmutableEntry<K,V>`

File: Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserUtil.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            "win32", "jxbrowser-win32-6.16",
+            "win64", "jxbrowser-6.16",
+            "mac", "jxbrowser-mac-6.16",
+            "linux64", "jxbrowser-linux64-6.16",
+            "linux32", "jxbrowser-linux64-6.16"
+            );
+
```
Comment (wezhang) [inline Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserUtil.java]: May the JxBrowser factory method be added?

Comment (djlan) [reply Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserUtil.java]: Ok, I will add all of them and send out a new iteration

---
In reply to: [154283559](https://github.com/Microsoft/azure-tools-for-java/pull/1279#discussion_r154283559) [](ancestors = 154283559)

File: Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserUtil.java (hunk 2 above)
Comment (djlan) [reply Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserUtil.java]: Use ImmutableMap for now

---
In reply to: [154283480](https://github.com/Microsoft/azure-tools-for-java/pull/1279#discussion_r154283480) [](ancestors = 154283480)

File: Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserUtil.java (hunk 3 above)
Comment (djlan) [reply Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserUtil.java]: Add simple method first. Will add more.

---
In reply to: [154285064](https://github.com/Microsoft/azure-tools-for-java/pull/1279#discussion_r154285064) [](ancestors = 154285064,154283559)

File: Utils/azuretools-core/pom.xml [hunk 4]
```diff
@@ -35,6 +35,12 @@
     <properties>
         <artifact.name>${project.artifactId}-${project.version}.jar</artifact.name>
     </properties>
+    <repositories>
+        <repository>
+            <id>com.teamdev</id>
+            <url>http://maven.teamdev.com/repository/products</url>
+        </repository>
+    </repositories>
```
Comment (djlan) [inline Utils/azuretools-core/pom.xml]: Will remove this after moving to download

File: Utils/azuretools-core/pom.xml [hunk 5]
```diff
@@ -184,5 +190,24 @@
...
+        <dependency>
+            <groupId>com.teamdev.jxbrowser</groupId>
+            <artifactId>jxbrowser-mac</artifactId>
+            <version>6.16</version>
+        </dependency>
+        <dependency>
+            <groupId>com.teamdev.jxbrowser</groupId>
+            <artifactId>jxbrowser-linux64</artifactId>
+            <version>6.16</version>
+        </dependency>
```
Comment (djlan) [inline Utils/azuretools-core/pom.xml]: Will remove this after moving to download

**Pull request merged**